src/webview-js/toc-editor/toc-editor.jsx

```diff
@@ -0,0 +1,155 @@
+import { h, Fragment, render } from 'preact';
+import { useEffect, useState } from 'preact/hooks';
+import 'react-sortable-tree/style.css';
+import SortableTree from 'react-sortable-tree';
+
+const vscode = acquireVsCodeApi();
+const nodeType = 'toc-element'
+
+const removeExpanded = (obj) => {
+  if (typeof obj !== 'object') {
+    return obj
+  }
+  if (obj instanceof Array) {
+    const copy = []
+    for (const subobj of obj) {
+      copy.push(removeExpanded(subobj))
+    }
+    return copy
+  }
+  const { expanded, ...copy } = obj
+  copy.children = removeExpanded(copy.children)
+  return copy
+}
+
+const ContentTree = (props) => {
+  const modifiesStateName = props.modifiesStateName
+  const [data, setData] = useState(props.data);
+
+  const handleChange = (newChildren) => {
+    const { treesData, selectionIndices } = vscode.getState()
+
+    const newData = { ...data, ...{children: newChildren} }
+    const oldStructure = JSON.stringify(removeExpanded(data.children))
+    const newStructure = JSON.stringify(removeExpanded(newChildren))
```

Comment:
Are these guaranteed to be serialized the same? I recall an npm package that provides repeatable JSON but maybe that is overkill since if they are structurally the same it's just one extra `postMessage` call.